### PR TITLE
Improve HA form default support

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Use default as placeholder for not required fields.
Don't reset field to 0 or min when empty.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-selector/ha-selector-number.ts [hunk 1]
```diff
@@ -81,17 +81,11 @@ export class HaNumberSelector extends LitElement {
...
-    return this.value ?? (this.selector.number?.min || 0);
-  }
-
   private _handleInputChange(ev) {
     ev.stopPropagation();
     const value =
       ev.target.value === "" || isNaN(ev.target.value)
-        ? this.required
-          ? this.selector.number?.min || 0
-          : undefined
```
Comment (piitaya) [inline src/components/ha-selector/ha-selector-number.ts:94]: @bramkragten Did you known why we have this ?

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:94]: If the consumer of the selector handles required correctly I think I agree...

File: src/components/ha-selector/ha-selector-number.ts [hunk 2]
```diff
@@ -81,17 +81,11 @@ export class HaNumberSelector extends LitElement {
...
+        ? ""
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-number.ts:88]: Why change this to an empty string instead of `undefined`?

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:88]: @piitaya

Comment (piitaya) [reply src/components/ha-selector/ha-selector-number.ts]: Changed

**Review by bramkragten — APPROVED** [review]

**Pull request merged**